packages/node_modules/@ciscospark/internal-plugin-locus/src/locus.js

```diff
@@ -397,6 +425,74 @@ const Locus = SparkPlugin.extend({
     return LESS_THAN;
   },
 
+  /**
+   * Merges two locus DTOs (for the same locus)
+   *
+   * @param {Types~Locus} current
+   * @param {Types~Locus|Types~LocusDelta} incoming
+   * @returns {Type~Locus}
+   */
+  merge(current, incoming) {
+    // if incoming is not a delta event, treat it as a new full locus.
+    if (!incoming.baseSequence) {
+      return incoming;
+    }
+
+    const next = cloneDeep(current);
+
+    // 1. All non-null elements in the delta event except the "baseSequence" and
+    // the "participants" collection should be used to replace their existing
+    // values.
+    Object.keys(incoming).forEach((key) => {
+      if (key === `baseSequence` || key === `participants`) {
+        return;
+      }
+
+      next[key] = incoming[key] || next[key];
+    });
+
+    // 2. The "baseSequence" in the delta event can be discarded (it doesn't
+    // need to be maintained in the local working copy).
+
+    if (incoming.participants || incoming.participants.length) {
+      const toRemove = new Set();
+      const toUpsert = new Map();
+
+      incoming.participants.forEach((p) => {
+        if (p.removed) {
+          // Elements of the delta event's "participants" list with the
+          // attribute `removed=true` should be removed from the working copy's
+          // "participants" collection.
+          toRemove.add(p.url);
+        }
+        else {
+          // Elements of the delta events "participants" list that are absent
+          // from the local working copy should be added to that collection.
+          toUpsert.set(p.url, p);
+        }
+      });
+
+      // The "participants" collection in the delta event should be merged with
```

Comment:
Thanks for the detailed notes in this one!

Reply:
haha, i pretty much copied them out of the spec. I'm not entirely sure i understand how this works :)